### PR TITLE
pg_upgrade: improve AO query performance for info_rels

### PR DESCRIPTION
**Pull request by jchampio**
The query to obtain AO auxiliary catalog names is relatively expensive compared to the other aux table queries, and it was being performed once for every AO table. Consolidate all calls into a single query, and manually join the results with the other relation info by relid.

With this change, the runtime for `--check` for one real-world customer schema improved from >24 hours to 9 minutes.

This PR also contains a patch for the pg_upgrade regression suite; see commit message for that.

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] ~Document changes~
- [ ] ~Communicate in the mailing list if needed~
- [x] Pass `make installcheck`
- [x] Review a PR in return to support the community

### CODE REVIEW

File: contrib/pg_upgrade/info.c [hunk 1]
```diff
@@ -727,46 +776,36 @@ get_rel_infos(ClusterInfo *cluster, DbInfo *dbinfo)
...
-
-			segrel = pg_strdup(PQgetvalue(aores, 0, PQfnumber(aores, "segrel")));
-			visimaprel = pg_strdup(PQgetvalue(aores, 0, PQfnumber(aores, "visimaprel")));
-			if (!PQgetisnull(aores, 0, PQfnumber(aores, "blkdirrel")))
-				blkdirrel = pg_strdup(PQgetvalue(aores, 0, PQfnumber(aores, "blkdirrel")));
+			// Since res and aorels are both sorted by relation OID, the next
+			// row in aorels should usually be exactly the relation we're
+			// looking for. The exception is that some AO tables aren't included
+			// in info_rels (e.g. append-only materialized views), so we have to
+			// handle extras in the list.
```
Comment (jchampio) [inline contrib/pg_upgrade/info.c:790]: This needs to be changed to C-style comments, sorry. Too much Go recently 😅

**Comment by kalensk** [comment]
Looks good to me, but I'll let other's with a bit more expertise/experience weigh in as well.

**Comment by asimrp** [comment]
24 hours to 9 minutes is still insanely surprising!

**Comment by jchampio** [comment]
Thanks for the review!

**Pull request merged**